api/src/main/java/org/apache/iceberg/view/ViewVersion.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.view;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * A version of the view at a point in time.
+ *
+ * <p>A version consists of a view metadata file.
+ *
+ * <p>Versions are created by view operations, like Create and Replace.
+ */
+public interface ViewVersion {
+
+  /** Returns this version's id. Version ids are monotonically increasing */
+  long versionId();
+
+  /** Returns this version's parent id or null if there is no parent */
+  Long parentId();
```

Comment:
What does it mean to have a parent version? If it is just the order in which view versions are created, can't we get that by looking at view with `versionId - 1`?

Reply:
You are right. Will we support rollback? If yes, the parent-child relation makes a little bit sense, for maintaining a lineage tree.

However, since nothing is shared between 2 views, "rollback" can be treated as a "replace" with all representations re-computed.

Reply:
I don't think parent makes sense because each view definition is independent. Maybe history matters, but we have history for that. Let's remove parent ID.